server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1186,6 +1227,431 @@ protected void pushToUpdateQueue(final List<NetworkVO> networks) throws Interrup
+            command.setAccessDetail(NetworkElementCommand.ROUTER_IP, controlIP);
+            command.setAccessDetail(NetworkElementCommand.ROUTER_NAME, router.getInstanceName());
+            try {
+                final Answer answer = _agentMgr.easySend(router.getHostId(), command);
+
+                if (answer == null) {
+                    s_logger.warn("Unable to fetch monitoring results data from router " + router.getHostName());
+                    return null;
+                }
+                if (answer instanceof GetRouterMonitorResultsAnswer) {
+                    return (GetRouterMonitorResultsAnswer) answer;
+                } else {
+                    s_logger.warn("Unable to fetch health checks results to router " + router.getHostName() + " Received answer " + answer.getDetails());
+                    return new GetRouterMonitorResultsAnswer(command, false, null, answer.getDetails());
+                }
+            } catch (final Exception e) {
+                s_logger.warn("Error while collecting alerts from router: " + router.getInstanceName(), e);
+                return null;
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public boolean performRouterHealthChecks(long routerId) {
+        DomainRouterVO router = _routerDao.findById(routerId);
+
+        if (router == null) {
+            throw new CloudRuntimeException("Unable to find router with id " + routerId);
+        }
+
+        if (!RouterHealthChecksEnabled.valueIn(router.getDataCenterId())) {
+            throw new CloudRuntimeException("Router health checks are not enabled in cluster router: " + router);
+        }
+
+        s_logger.info("Running health check results for router " + router.getUuid());
+
+        // Step 1: Update health check data on router.
+        if (!updateRouterHealthCheckData(router)) {
+            s_logger.warn("Unable to update health check data for fresh run successfully for router: " + router);
+            return false;
+        }
+        s_logger.info("Updated health check data for fresh run successfully for router: " + router);
+
+        // Step 2: Perform and retrieve health checks on router
+        s_logger.info("Retrieving results for fresh health check execution for router " + router.getUuid());
+        GetRouterMonitorResultsAnswer answer = fetchAndUpdateRouterHealthChecks(router, true);
+
+        // Step 3: Update health checks values in database
+        if (answer == null || !answer.getResult()) {
+            updateRouterConnectivityHealthCheck(routerId, false,
+                    answer == null ? "Communication failed " : "Failed to fetch results with details: " + answer.getDetails());
+        } else {
+            updateRouterConnectivityHealthCheck(routerId, true, "Successfully fetched data");
+            parseAndMergeDbHealthChecks(routerId, answer.getMonitoringResults());
+        }
```

Comment:
the methods called are also logging the warnings and the info seems excessive in some cases. I like good logging, so don't be discouraged by this comment. maybe simplify?

Reply:
Tried to simplify. Let me know your thoughts.